### ISSUE TITLE
multilayer perceptron output classes error

### ISSUE THREAD

**Issue by diouck** [keras-team/keras#14655]
Hello,I am trying to set up a model for classifying satellite data into 3 classes. The model works but during the prediction I only have 2 output classes. Did you encounter this problem? Here is my code

 
 `

X = img[roi > 0, :]  # include 8th band, which is Fmask, for now
Y = roi[roi > 0]

# encode class values as integers
encoder = LabelEncoder()
encoder.fit(Y)
encoded_Y = encoder.transform(Y)
# convert integers to dummy variables (i.e. one hot encoded)
dummy_y = np_utils.to_categorical(encoded_Y)

print('Our X matrix is sized: {sz}'.format(sz=X.shape))
print('Our Y array is sized: {sz}'.format(sz=dummy_y.shape))

# Split the data up in train and test sets
X_train, X_test, Y_train, Y_test = train_test_split(X,dummy_y, test_size=0.20, random_state=42)
print(X_train.shape,Y_train.shape,X_test.shape,Y_test.shape)

# Scale the train & test data set
X_train = preprocessing.scale(X_train)
X_test = preprocessing.scale(X_test)

model = Sequential()
model.add(Dense(64, input_dim=4, activation='relu'))
model.add(Dense(3, activation='softmax'))

# Compile model
from keras.optimizers import SGD
sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)

model.compile(loss='categorical_crossentropy', optimizer='sgd', metrics=['accuracy'])

# Fit model                
history=model.fit(X_train, 
          Y_train,
          epochs=2, 
          batch_size=100, 
          validation_split = 0.2,
          verbose=1,
          )
# list all data in history
print(history.history.keys())

y_test_arg=np.argmax(Y_test,axis=1)
Y_pred = np.argmax(model.predict(X_test),axis=1)
print('Confusion Matrix')
print(confusion_matrix(y_test_arg, Y_pred))

#Évaluation du réseau
loss, accuracy = model.evaluate(X_test,Y_test)
print("Accuracy", accuracy)

#Matrix
[[2222    1   12]
 [  23 3561  682]
 [  11  257 3876]]

`
